Makefile

```diff
@@ -1,32 +1,49 @@
 CRATE_NAME := x25519-chacha20poly1305
-
+SCOPE_NAME := entropyxyz
 default :: build-nodejs
 
+
+# Builds a JS Module for bundler with glue for the compiled WASM.
+build-bundler ::
+	wasm-pack build --target bundler --out-dir pkg/bundler --out-name $(CRATE_NAME)-bundler --scope $(SCOPE_NAME)
+
 # Builds a JS Module for nodejs with glue for the compiled WASM.
 build-nodejs ::
-	wasm-pack build --target nodejs
-
+	wasm-pack build --target nodejs --out-dir pkg/nodejs --out-name $(CRATE_NAME)-nodejs --scope $(SCOPE_NAME)
 
 # Builds a JS Module for web with glue for the compiled WASM.
 build-web ::
-	wasm-pack build --target web
+	wasm-pack build --target web --out-dir pkg/web --out-name $(CRATE_NAME)-web --scope $(SCOPE_NAME)
+
+
 
-# Links the local wasm/js library with npm for testing locally.
-link-local :: build-nodejs
-	cd pkg && npm link
+# Links the local bundler wasm/js library with npm for testing locally.
+link-local-bundler:: build-bundler
+	cd pkg/bundler && npm link
+
+# Links the local nodejs wasm/js library with npm for testing locally.
+link-local-nodejs :: build-nodejs
+	cd pkg/nodejs && npm link
+
+# Links the local web wasm/js library with npm for testing locally.
+link-local-web :: build-web
+	cd pkg/web && npm link
 
 # Links the example frontend project with the locally built wasm/js package.
-link :: link-local
-	cd example && npm link $(CRATE_NAME) 
+link-example-bundler :: link-local-bundler
+	cd example && npm link $(CRATE_NAME)/bundler
 
-# Another build option for compiling to webpack, builds a typescript library around the WASM for use
-# with npm.
-build-bundler ::
```

Comment:
I ended up adding bundler commands for each `build-*`, `link-local-*` and `link-example-*` since it was defined in a build command.

Reply:
👍

Reply:
Regarding make troubles:

For certain cmds like (ls, cd, mv) you specify as make cmds so $(CD) $(MV), etc...

In addition to that, make doesn't know where/what shell you want to use, adding the following to the top will fix that issue

```
SHELL := /bin/bash
```

If you change directory it's best to use make -C and place a new Makefile in the sub directory.